libkbfs/folder_branch_ops.go

```diff
@@ -4806,6 +4806,12 @@ func (fbo *folderBranchOps) handleMDFlush(ctx context.Context, bid BranchID,
 	fbo.mdWriterLock.Lock(lState)
 	defer fbo.mdWriterLock.Unlock(lState)
 
+	if err := isArchivableMDOrError(rmd.ReadOnly()); err != nil {
```

Comment:
Seems like this can go above the lockState creation and lock grab, to save a little time in the error case.